src/nvim/os/fs.c

```diff
@@ -114,17 +114,19 @@ bool os_can_exe(const char_u *name, char_u **abspath)
 static bool is_executable(const char_u *name)
   FUNC_ATTR_NONNULL_ALL
 {
+#ifdef WIN32
+    // Windows does not have an executable bit.
+    // Instead check if the file exists and is readable.
+    return os_file_is_readable((char *) name);
+#else
```

Comment:
Sorry, I should have read the docs more carefully. We do need to use `S_ISREG` because that checks that the file is not a directory.